tests/unit/test_retry.py

```diff
@@ -129,6 +129,25 @@ def test_retry_target_non_retryable_error(utcnow, sleep):
     sleep.assert_not_called()
 
 
+@mock.patch("asyncio.sleep", autospec=True)
+@mock.patch(
+    "google.api_core.datetime_helpers.utcnow",
+    return_value=datetime.datetime.min,
+    autospec=True,
+)
+@pytest.mark.asyncio
+async def test_retry_target_warning_for_retry(utcnow, sleep):
+    predicate = retry.if_exception_type(ValueError)
+    target = mock.AsyncMock(spec=["__call__"])
+
+    with pytest.warns(Warning) as exc_info:
+        retry.retry_target(target, predicate, range(10), None)
```

Comment:
IIUC, `predicate` is not used, right? So I would suggest one of (a) adding a comment in line 140 saying that this is just a filler that shouldn't affect the test, or (b) setting  it to `None`, or (c) doing each of (a) and (b) in separate test cases, to show `predicate` doesn't affect things.

I think just (a) would be fine, though.